modules/ghlint-api/src/main/kotlin/net/twisterrob/ghlint/model/Permission.kt

```diff
@@ -17,147 +52,36 @@ public enum class Access: Comparable<Access> {
-				RepositoryProjects.Name -> RepositoryProjects(access)
-				SecurityEvents.Name -> SecurityEvents(access)
-				Statuses.Name -> Statuses(access)
-				else -> Undefined(access)
-			}
-		}
-	}
-	public data class Actions(override val access: Access) : Permission(access) {
-		public companion object {
-			public const val Name: String = "actions"
-		}
-
-		public override val name: String = Name
-	}
-
-	public data class Attestations(override val access: Access) : Permission(access) {
-		public companion object {
-			public const val Name: String = "attestations"
-		}
-
-		public override val name: String = Name
-	}
-
-	public data class Checks(override val access: Access) : Permission(access) {
-		public companion object {
-			public const val Name: String = "checks"
-		}
-
-		public override val name: String = Name
-	}
-
-	public data class Contents(override val access: Access) : Permission(access) {
-		public companion object {
-			public const val Name: String = "contents"
-		}
-
-		public override val name: String = Name
-	}
-
-	public data class Deployments(override val access: Access) : Permission(access) {
-		public companion object {
-			public const val Name: String = "deployments"
-		}
-
-		public override val name: String = Name
-	}
-
-	public data class IdToken(override val access: Access) : Permission(access) {
-		public companion object {
-			public const val Name: String = "id-token"
-		}
-
-		public override val name: String = Name
-	}
-
-	public data class Issues(override val access: Access) : Permission(access) {
-		public companion object {
-			public const val Name: String = "issues"
-		}
-
-		public override val name: String = Name
-	}
-
-	public data class Metadata(override val access: Access) : Permission(access) {
-		public companion object {
-			public const val Name: String = "metadata"
-		}
-
-		public override val name: String = Name
-	}
-
-	public data class Packages(override val access: Access) : Permission(access) {
-		public companion object {
-			public const val Name: String = "packages"
-		}
-
-		public override val name: String = Name
-	}
-
-	public data class Pages(override val access: Access) : Permission(access) {
-		public companion object {
-			public const val Name: String = "pages"
-		}
-
-		public override val name: String = Name
-	}
-
-	public data class PullRequests(override val access: Access) : Permission(access) {
-		public companion object {
-			public const val Name: String = "pull-requests"
-		}
-
-		public override val name: String = Name
-	}
-
-	public data class RepositoryProjects(override val access: Access) : Permission(access) {
-		public companion object {
-			public const val Name: String = "repository-projects"
-		}
-
-		public override val name: String = Name
+public data class Scope(val permission: Permission, val access: Access) {
+	override fun toString(): String {
+		return "`${permission.name.lowercase(Locale.getDefault())}: ${access.name.lowercase(Locale.getDefault())}`"
```

Comment:
pull-requests, id-token 😶

(I think PULL_REQUESTS("pull-requests") is ok for the enum, at least there's a place for the constant in the model, and rule authors and even SnakePermissions can use that.)

Reply:
Same for Access, just cause R8 (code is minified, very likely these classes and their constants will be named "a" / "b" at runtime.